akka-remote/src/test/scala/akka/remote/artery/ArteryMultiNodeSpec.scala

```diff
@@ -62,9 +73,32 @@ abstract class ArteryMultiNodeSpec(config: Config) extends AkkaSpec(config.withF
     remoteSystem
   }
 
+  // keep track of failure so that we can print flight recorder output on failures
+  private var failed = false
+  override protected def withFixture(test: NoArgTest): Outcome = {
+    val out = super.withFixture(test)
+    if (!out.isSucceeded) failed = true
+    out
+  }
+
+  override protected def beforeTermination(): Unit = {
+    handleFlightRecorderFile()
+  }
+
   override def afterTermination(): Unit = {
     remoteSystems.foreach(sys ⇒ shutdown(sys))
     remoteSystems = Vector.empty
   }
 
+  private def handleFlightRecorderFile(): Unit = {
+    if (Files.exists(flightRecorderFile)) {
+      if (failed) {
+        // logger may not be alive anymore so we have to use stdout here
+        println("Flight recorder dump:")
```

Comment:
I started out with "=== Flight recorder dump ===" but it looked like crap, The dump is so different anyways that it stands out well.